src/main/java/github/scarsz/discordsrv/linking/AccountSystem.java

```diff
@@ -0,0 +1,43 @@
+/*-
+ * LICENSE
+ * DiscordSRV
+ * -------------
+ * Copyright (C) 2016 - 2021 Austin "Scarsz" Shapiro
+ * -------------
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-3.0.html>.
+ * END
+ */
+
+package github.scarsz.discordsrv.linking;
+
+import github.scarsz.discordsrv.linking.store.CodeStore;
+import github.scarsz.discordsrv.linking.store.DiscordAccountStore;
+import github.scarsz.discordsrv.linking.store.MinecraftAccountStore;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.UUID;
+
+public interface AccountSystem extends DiscordAccountStore, MinecraftAccountStore, CodeStore {
+
+    @NotNull AccountLinkResult process(String code, String discordId);
+    int getLinkCount();
+
+    default void link(UUID playerUuid, String discordId) {
```

Comment:
There doesn't seem to be a mechanism for this failing/throwing anything even if the backend is mcauth? (would be useful as this is user facing)

Reply:
MCAuth isn't a valid backend currently